### ISSUE TITLE
Can gookie pull from a private git repo using SSH credentials?

### ISSUE THREAD

**Issue by oxymor0n** [hoxxep/Gookie#4]
Hello,

First of all, thanks for making this. It's a pretty nifty CI tool.

So I have a private repo that I want to deploy using Gookie. However, I don't want to pull via the usual HTTP URL, because then I'd have to cache my github password in plaintext for the continuous integration to work. The usual way to deal with this is to pull via SSH instead, but as it stands Gookie fails when the url config is in the ssh format (e.g. git@github.com:wiringtheworld/Gookie.git)

Is there any plan to support this? If not, can you point me to the relevant section of the code so that I can implement this feature? 

Thanks,
Will

**Comment by hoxxep** [comment]
Hey, cheers! 

The URL that's in the Gookie config is from the webhook request, so the https format is the one that should be used. When you want to pull via SSH, you configure that in your local git repository's remotes, `git remote set-url origin ssh://git@github.com/user/repo` and then you should be able to run `git pull` without any issues, provided you have used an unencrypted SSH key.

So your local git repository remotes and SSH key are separate from the Gookie service, as all Gookie does is receive a webhook request, then run a custom script/command. The script itself (and set up of the repo you're pulling) are in effect separate from Gookie, so it doesn't have to know about any auth setup provided it does not ask for a password.

TL;DR: The `url` in `config.json` is for the webhook, to validate the repository it has been received for, and should be the full github url of the repository that you would visit in a browser. Setting up `git pull` via SSH for private repositories is done in your repository; Gookie does not use/need this information as it simply runs the `git pull` command (or custom command)  in the same way a user would from a terminal.

Hope that helps, let me know if anything is unclear?

Cheers,
Liam

**Comment by oxymor0n** [comment]
Wow, that is very prompt and clear reply. Thanks so much for that!

Best,
Will
